### PR TITLE
A few more php 8.1 things

### PR DESCRIPTION
**Pull request by thirsch**
I'm currently comparing the current master to our fork with php 8.0 and 8.1 fixes and stumbled upon these changes.

### CODE REVIEW

File: lib/Doctrine/Hydrator/RecordDriver.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public function flush()
             $coll->takeSnapshot();
         }
         $this->_initializedRelations = null;
-        $this->_collections = null;
+        $this->_collections = array();
```
Comment (thirsch) [inline lib/Doctrine/Hydrator/RecordDriver.php:121]: This is not really related to 8.1. Flushing twice caused a warning due to the `$_collection` being set to `null`.

File: lib/Doctrine/Table.php [hunk 2]
```diff
@@ -1704,7 +1704,7 @@ public function findByDql($dql, $params = array(), $hydrationMode = null)
      * Find records basing on a field.
      *
      * @param string $column            field for the WHERE clause
-     * @param string $value             prepared statement parameter
+     * @param string|array $value       prepared statement parameter
```
Comment (thirsch) [inline lib/Doctrine/Table.php:1707]: These two are not related to 8.1 as well, but it's nice in editors with code assist to know the correct types.

**Comment by JohannesTyra** [comment]
Thx @thePanz for merging the commit.

The release v1.4.0-RC1 is marked as "pre-release", is there a "final release" v.1.4.x marked as "latest" planed soon or are there any open issues / pull-requests?

**Comment by thePanz** [comment]
@JohannesTyra I tagged v1.4.0-RC2

The aim is to gather more feedback from users before releasing the v1.4.0
Do you face any issues with the current v1.4.0-RC2?

**Comment by JohannesTyra** [comment]
@thePanz Thx. I will check it asap.

**Comment by JohannesTyra** [comment]
@thePanz i found something.

Its on building the doctrine classes via console.

`
>> doctrine  generating form classes

PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in vendor/friendsofsymfony1/doctrine1/lib/Doctrine/Table.php on line 1182
`

***

Afterwords its IN the generated base classes und results in corrupted files.
(because the output is passed there)

**Comment by JohannesTyra** [comment]
@thePanz added an issue: https://github.com/FriendsOfSymfony1/doctrine1/issues/95

**Comment by thirsch** [comment]
> @thePanz i found something.
> 
> Its on building the doctrine classes via console.
> 
> `
> 
> > > doctrine  generating form classes
> 
> PHP Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in vendor/friendsofsymfony1/doctrine1/lib/Doctrine/Table.php on line 1182 `
> 
> Afterwords its IN the generated base classes und results in corrupted files. (because the output is passed there)

@thePanz as the fix for this is merged now, how about a release 1.4.0?

**Comment by thePanz** [comment]
Yup, why not :)
Will tag the release soon
